pkg/scm/git/git.go

```diff
@@ -217,84 +220,75 @@ func makePathAbsolute(source string) string {
 // expect git clone spec syntax; it also provides details if the file://
 // proto was explicitly specified, if we should use OS copy vs. the git
 // binary, and if a frag/ref has a bad format
-func ParseFile(source string) (details *FileProtoDetails, mods *URLMods) {
+func ParseFile(source string) (*FileProtoDetails, *URLMods, error) {
+	// Checking to see if the user included a "file://" in the call
 	protoSpecified := false
 	if strings.HasPrefix(source, "file://") && len(source) > 7 {
 		protoSpecified = true
 	}
 
-	// in each valid case, like the prior logic in scm.go did, we'll make the
-	// paths absolute and prepend file:// to the path which callers should
-	// switch to
+	refSpecified := false
+	path, ref := "", ""
+	if strings.LastIndex(source, "#") != -1 {
+		refSpecified = true
 
-	// if source, minus potential file:// prefix and #ref suffix,
-	// exists as is, denote and return
-	if doesExist(source) || doesExist(strings.SplitN(source, "#", 2)[0]) {
-		details = &FileProtoDetails{
-			FileExists:     true,
-			UseCopy:        useCopy(protoSpecified, source),
-			ProtoSpecified: protoSpecified,
-			BadRef:         false,
-		}
-		mods = &URLMods{
-			Scheme: "file",
-			Path:   makePathAbsolute(strings.TrimPrefix(source, "file://")),
-		}
-		return
+		segments := strings.SplitN(source, "#", 2)
+		path = segments[0]
+		ref = segments[1]
+	} else {
+		path = source
 	}
 
-	// need to see if this was a file://<valid file>#ref
-	if protoSpecified && strings.LastIndex(source, "#") != -1 {
-		segments := strings.SplitN(source, "#", 2)
-		// given last index check above, segement should
-		// be of len 2
-		path, ref := segments[0], segments[1]
-		// file does not exist, return bad
-		if !doesExist(path) {
-			details = &FileProtoDetails{
-				UseCopy:        false,
-				FileExists:     false,
-				ProtoSpecified: protoSpecified,
+	// in each valid case, like the prior logic in scm.go did, we'll make the
+	// paths absolute and prepend file:// to the path which callers should
+	// switch to
+	if doesExist(path) {
+		isValidGit, err := isValidGitRepository(path)
+		hasGit := hasGitBinary()
+		if err != nil {
+			details := &FileProtoDetails{
+				UseCopy:        !(isValidGit && hasGit),
+				FileExists:     true,
 				BadRef:         false,
+				ProtoSpecified: protoSpecified,
 			}
-			mods = nil
-			return
+			return details, nil, err
 		}
+
 		// if ref/frag bad, return bad
-		if !gitSSHURLPathRef.MatchString(ref) {
-			details = &FileProtoDetails{
-				UseCopy:        false,
+		if refSpecified && !gitSSHURLPathRef.MatchString(ref) {
+			details := &FileProtoDetails{
+				BadRef:         true,
 				FileExists:     true,
 				ProtoSpecified: protoSpecified,
-				BadRef:         true,
+				UseCopy:        !(isValidGit && hasGit),
```

Comment:
and then this can be UseCopy: false  since you've eliminated the "invalid git repo" and "missing git binary" cases above, if you make the change i suggested.